src/elf.rs

```diff
@@ -5,8 +5,10 @@ use object_trait::Object;
 pub struct Elf<'a>(xmas_elf::ElfFile<'a>);
 
 impl<'a> Object<'a> for Elf<'a> {
-    fn parse(input: &'a [u8]) -> Elf<'a> {
-        Elf(xmas_elf::ElfFile::new(input))
+    type Error = ();
+
+    fn parse(input: &'a [u8]) -> Result<Elf<'a>, Self::Error> {
+        Ok(Elf(xmas_elf::ElfFile::new(input)))
```

Comment:
Note that xmas_elf currently can assert here if the magic is wrong, and panic if input is too small. It currently uses `&'static str` for error in other places, so I guess that's what it'll eventually use here too.